webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js

```diff
@@ -43,15 +43,15 @@ const ComponentContentViewAddModal = ({
   useEffect(() => {
```

Comment:
We have a bad habit (especially in CV UI) of using effects to compute what should be rendered. 

https://react.dev/learn/you-might-not-need-an-effect

Could this useEffect be eliminated? Or at least remove a lot of the data normalization logic from it and do that during render?

Reply:
Interesting..I can see removing useMemo in the bulk modal cause we pass all details to render the modal as props so there aren't any API calls that need to propogate effects on success.

The single add/edit modal does send out a call to the API and useEffect looks at API return status. That seems necessary.

Reply:
Updated bulk modal..

Reply:
```
diff --git a/webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js b/webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js
index c09d09de43..7309f894f5 100644
--- a/webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js
+++ b/webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js
@@ -1,4 +1,4 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import {
   Flex, Modal, ModalVariant, Select, SelectVariant,
@@ -33,24 +33,20 @@ const ComponentContentViewAddModal = ({
     state => selectCVDetailError(state, componentCvId),
     shallowEqual,
   );
-  const [cvName, setCvName] = useState('');
-  const [options, setOptions] = useState([]);
   const [formLatest, setFormLatest] = useState(componentId ? latest : false);
   const [selected, setSelected] = useState(null);
   const [cvVersionSelectOpen, setCvVersionSelectOpen] = useState(false);
   const versionsLoading = componentStatus === STATUS.PENDING;
+  const cvName = componentDetails?.name ?? '';
+  const options = useMemo(() => (componentDetails?.versions ?? []).map(item => ({
+    value: item.id, label: __(`Version ${item.version}`), description: item.description, publishedAtWords: __(` (${item.published_at_words} ago)`),
+  })), [componentDetails?.versions]);
 
   useEffect(() => {
     if (!versionsLoading && componentDetails) {
-      const { name, versions } = componentDetails;
-      setCvName(name);
-      const opt = versions.map(item => ({
-        value: item.id, label: __(`Version ${item.version}`), description: item.description, publishedAtWords: __(` (${item.published_at_words} ago)`),
-      }));
-      setOptions(opt);
-      setSelected(componentVersionId || opt[0]?.value);
+      setSelected(componentVersionId || options[0]?.value);
     }
-  }, [componentDetails, componentStatus, componentError, versionsLoading, componentVersionId]);
+  }, [versionsLoading, componentDetails, componentVersionId, options]);
 
   const getAddParams = () => {
     if (formLatest) {
```

This simplifies the effect and removes 2 states. This is what I meant by remove the data normalization logic from the effect.

Reply:
Now that I think about it, we could remove the effect entirely if we move `componentVersionId || options[0]?.value` into the default value for `selected`:

```js
- const [selected, setSelected] = useState(null);
+ const [selected, setSelected] = useState(componentVersionId || options[0]?.value);
```

(you'd also have to move that line to after we declare `options`)